packages/http-server-csharp/eng/scripts/emit-scenarios.js

```diff
@@ -0,0 +1,278 @@
+const testScenarioPath = join(basePath, "test", "scenarios");
+const ignoreFilePath = join(testScenarioPath, ".testignore");
+
+const argv = yargs(hideBin(process.argv))
+  .option("input", {
+    alias: "i",
+    type: "string",
+    description: "Input spec folder",
+    default: join(basePath, "..", "http-specs", "specs"),
+  })
+  .option("build", {
+    type: "boolean",
+    describe: "Build the generated projects",
+    default: true,
+  })
+  .option("interactive", {
+    type: "boolean",
+    describe: "Enable interactive mode",
+    default: false,
+  })
+  .help().argv;
+
+const specDir = resolve(argv.input);
+
+// Initialize the port number
+let portNumber = 65000;
+
+// Remove the log directory if it exists.
+async function clearDirectory(dirRoot) {
+  if (await pathExists(dirRoot)) {
+    await rm(dirRoot, { recursive: true, force: true });
+  }
+}
+
+async function copySelectiveFiles(extension, sourceDir, targetDir) {
+  const files = await globby(extension, { cwd: sourceDir });
+  for (const file of files) {
+    const src = join(sourceDir, file);
+    const dest = join(targetDir, file);
+    await copy(src, dest);
+  }
+}
+
+async function compileSpec(file, options) {
+  const relativePath = file;
+  const fullPath = resolve(specDir, relativePath);
+  const { build, interactive } = options;
+  const patchFileDir = join(testScenarioPath, dirname(relativePath));
+  const outputDir = join(testScenarioPath, "generated", dirname(relativePath));
+  const logDir = join(logDirRoot, dirname(relativePath));
+
+  let spinner;
+  if (interactive) {
+    spinner = ora({ text: `Processing: ${relativePath}`, color: "cyan" }).start();
+  }
+
+  try {
+    if (await pathExists(outputDir)) {
+      if (spinner) spinner.text = `Clearing directory: ${outputDir}`;
+      await rm(outputDir, { recursive: true, force: true });
+    }
+    if (spinner) spinner.text = `Creating directory: ${outputDir}`;
+    await mkdir(outputDir, { recursive: true });
+
+    // Increment the port number for each folder
+    portNumber++;
+
+    // Compile the spec and generate server code
+    if (spinner) spinner.text = `Generating csharp server code: ${relativePath}`;
+    await execa("npx", [
+      "tsp",
+      "compile",
+      fullPath,
+      "--emit",
+      resolve(import.meta.dirname, "../.."),
+      "--config",
+      tspConfig,
+      "--output-dir",
+      join(outputDir, "generated"),
+      "--arg",
+      `service-port-http=${portNumber.toString()}`,
+      "--arg",
+      `service-port-https=${(portNumber + 500).toString()}`,
+    ]);
+
+    if (spinner) spinner.text = `Formatting with Prettier: ${relativePath}`;
+    await execa("npx", ["prettier", outputDir, "--write"]);
+
+    if (build) {
+      if (spinner) spinner.text = `Building project: ${relativePath}`;
+      await execa("dotnet", ["build"], { cwd: outputDir });
+    }
+
+    if (await pathExists(patchFileDir)) {
+      const mockDir = join(outputDir, "mocks");
```

Comment:
Is this where we copy in the real bl implementation?

Reply:
I assumed the mocks goes there. But I can adjust the copy logic to copy any files in the scenario folder recursively. Let's discuss tomorrow.